src/character.cpp

```diff
@@ -11822,6 +11883,75 @@ item &Character::item_with_best_of_quality( const quality_id &qid )
     return null_item_reference();
 }
 
+int Character::limb_health_movecost_modifier() const
+{
+    // the best hp score of all legs. 1 is full health
+    float leg_hi = 0.0f;
+    // the second best hp score of all legs. 1 is full health.
+    float leg_lo = 0.0f;
+
+    for( const bodypart_id &part : get_all_body_parts() ) {
+        if( part->limb_type == body_part_type::type::leg ) {
+            const float cur_leg = static_cast<float>( get_part_hp_cur( part ) ) /
+                                  static_cast<float>( get_part_hp_max( part ) );
+            if( cur_leg > leg_hi ) {
+                leg_lo = leg_hi;
+                leg_hi = cur_leg;
+            }
+        }
+        if( leg_hi == 1.0f && leg_lo == 1.0f ) {
+            break;
+        }
+    }
+
+    return 50 * ( 1 - std::sqrt( leg_hi ) ) +
+           50 * ( 1 - std::sqrt( leg_lo ) );
+}
+
+int Character::foot_encumbrance_movecost_modifier() const
+{
+    // the lowest encumbered leg
+    cata::optional<int> encumb_leg_lo = cata::nullopt;
+    // the second lowest encumbered leg
+    cata::optional<int> encumb_leg_hi = cata::nullopt;
+    // the lowest encumbered foot
+    cata::optional<int> encumb_foot_lo = cata::nullopt;
+    // the second lowest encumbered foot
+    cata::optional<int> encumb_foot_hi = cata::nullopt;
+
+    for( const bodypart_id &part : get_all_body_parts() ) {
+        if( part->limb_type == body_part_type::type::leg ) {
+            const int leg_encumb = encumb( part );
+            if( !encumb_leg_lo ) {
+                encumb_leg_lo = leg_encumb;
+            } else if( *encumb_leg_lo > leg_encumb ) {
+                encumb_leg_hi = encumb_leg_lo;
+                encumb_leg_hi = leg_encumb;
+            }
+        } else if( part->limb_type == body_part_type::type::foot ) {
+            const int foot_encumb = encumb( part );
+            if( !encumb_foot_lo ) {
+                encumb_foot_lo = foot_encumb;
+            } else if( *encumb_foot_lo > foot_encumb ) {
+                encumb_foot_hi = encumb_foot_lo;
+                encumb_foot_hi = foot_encumb;
+            }
+        }
+    }
+
+    /**
+     * since we're doing a min function but still want to zero out encumbrance,
+     * optional seems the easiest way to go here rather than using INT_MAX which
+     * would do the same thing.
+     */
+    const int encumb_feet = !!encumb_foot_hi ? *encumb_foot_hi : 0 +
+                            !!encumb_foot_lo ? *encumb_foot_lo : 0;
+    const int encumb_legs = !!encumb_leg_hi ? *encumb_leg_hi : 0 +
+                            !!encumb_leg_lo ? *encumb_leg_lo : 0;
```

Comment:
A better way might be
```suggestion
    const int encumb_feet = encumb_foot_hi.value_or( 0 ) + encumb_foot_lo.value_or( 0 );
    const int encumb_legs = encumb_leg_hi.value_or( 0 ) + encumb_leg_lo.value_or( 0 );
```